### PR TITLE
Updates components to rounded rectangle style

### LINKED ISSUE #514: Dropdown updates
- [ ] Updated rounded corner values for each size to match button values
- [ ] Updated heights of the actual inputs to match button values

https://www.figma.com/file/g8Z0q6TMPYDq6zXh9Y7LWD/%F0%9F%8C%8E-Universal?type=design&node-id=195%3A7599&mode=design&t=VHhtLpJltP9ac2IA-1

### LINKED ISSUE #567: Text field updates
- [ ] Updated rounded corner values for each size to match button values
- [ ] Updated heights of the actual inputs to match button values

https://www.figma.com/file/g8Z0q6TMPYDq6zXh9Y7LWD/%F0%9F%8C%8E-Universal?type=design&node-id=195%3A7599&mode=design&t=VHhtLpJltP9ac2IA-1

### PR DESCRIPTION
**Pull request by AlanBreck**
Resolves https://github.com/brave/leo/issues/510
Resolves https://github.com/brave/leo/issues/511
Resolves https://github.com/brave/leo/issues/528
Resolves #567
Resolves #514

NOTE: this should be rebased and not squashed.

### CODE REVIEW

File: src/components/segmentedControl/segmentedControl.svelte
Comment (AlanBreck) [inline src/components/segmentedControl/segmentedControl.svelte:51]: This may not be a great way to do this, but I'm running into a problem where the initial width isn't _quite_ large enough, so the theory here is to repeatedly request an animation for reasonably long enough period to assume the final width is correct.

**Review by aguscruiz — COMMENTED** [review]
So cool to see this, couple comments:

- [x] The hero button has an old gradient on hover, instead of darkening the current gradient with the calculation we had set up.
- [ ] About the border radius, I'm gonna leave a link here for discussing later. What I did on the buttons is this weird calculation to smooth the borders (in Figma) but that doesn't come across in straight css it seems. Let me know later, maybe in tomorrow's call, if this is something we can do https://css-houdini.rocks/smooth-corners/ (Search squircle in css for more). 

It's a silly thing to notice, but I promise it makes all the difference on how sleek they look

The rest looks great

**Review by aguscruiz — COMMENTED** [review]
Looks like the Jumbo buttons are much bigger than what's intended.
In Figma they're 60x60. 

image

-------------------

Same is happening with the rest of the sizes, they seem to be way bigger

**Comment by AlanBreck** [comment]
@aguscruiz, thanks for catching! This should now be resolved. Note that with the incorrect padding removed on icon-only buttons, but hover effect no longer looked good on `plain`/`plain-faint` buttons, so I removed.

**Review by aguscruiz — COMMENTED** [review]
Sorry, one more comment I didn't catch before. I think the non-plain buttons have their transitions to hover animated, and the plain variants don't 🥺

**Comment by AlanBreck** [comment]
> Sorry, one more comment I didn't catch before. I think the non-plain buttons have their transitions to hover animated, and the plain variants don't 🥺

`plain` buttons are actually currently transitioning the same as non-plain, but `plain-faint` is not. Working on a fix.

File: src/components/button/button.svelte
Comment (AlanBreck) [inline src/components/button/button.svelte:412]: Note that since `plain-faint` relies on `currentColor`, transitions are working, so I'm using a pseudo element instead in order to transition the opacity of the element instead of the background color.

**Comment by AlanBreck** [comment]
Button size when an `<a>` is really tall.

Sizes of inputs and buttons do not match.

**Comment by fallaciousreasoning** [comment]
@petemill @aguscruiz we should decide how we want to land this in Brave Core. Do we need to update all the native buttons there before this lands? Otherwise it's going to be a bit odd having square buttons some places and round ones in others

**Comment by aguscruiz** [comment]
Ideally, if we can somehow change all the native buttons to squircle too, that'd be a huge win. But there are also other changes to be made, like the overall use of orange being replaced with blurple. 

But, we can do it incrementally. let me know what you think will be better

**Comment by AlanBreck** [comment]
@fallaciousreasoning, by updating the `formItem` component for the `Input`, did that also update the `Dropdown`s, and therefore resolve https://github.com/brave/leo/issues/514?

**Comment by fallaciousreasoning** [comment]
Should have, yeah :smile:

**Review by fallaciousreasoning — APPROVED** [review]

**Pull request merged**
